src/openforms/emails/digest.py

```diff
@@ -257,3 +296,119 @@ def collect_invalid_certificates() -> list[InvalidCertificate]:
+        # a more general error and the link to the broken form to investigate
+        serializer = plugin.configuration_options(
+            data=backend_options,
+            context={"validate_business_logic": True},
+        )
+
+        try:
+            serializer.is_valid(raise_exception=True)
+        except serializers.ValidationError:
+            invalid_registration_backends.append(
+                InvalidRegistrationBackend(
+                    config_name=plugin.verbose_name,
+                    exception_message=_(
+                        "Invalid registration backend configuration detected"
+                    ),
+                    form_name=form_name,
+                    form_id=form.id,
+                )
+            )
+
+    return invalid_registration_backends
+
+
+def collect_invalid_logic_rules() -> list[InvalidLogicRule]:
+    forms = Form.objects.live().iterator()
+    static_variables = {
+        var.key: {"source": var.source, "type": var.data_type}
+        for var in get_static_variables()
+    }
+
+    invalid_logic_rules = []
+    for form in forms:
+        form_variables = {
+            form_variable.key: {
+                "source": form_variable.source,
+                "type": form_variable.data_type,
+            }
+            for form_variable in form.formvariable_set.all()
+        }
+
+        all_keys = list(static_variables) + list(form_variables)
+
+        form_logics = FormLogic.objects.filter(form=form)
+
+        form_logics_vars = []
+        for logic in form_logics:
+            for var in introspect_json_logic(logic.json_logic_trigger).get_input_keys():
+                form_logics_vars.append(var)
+            for action in logic.actions:
+                if action_var := action.get("variable"):
+                    form_logics_vars.append(InputVar(key=action_var))
+
+        for var in set(form_logics_vars):
+            outer, *nested = var.key.split(".")
+            if var.key not in all_keys:
+                expected_container_types = {FormVariableDataTypes.object}
+                parent_var = form_variables.get(outer) or static_variables.get(outer)
+
+                if parent_var is not None:
+                    if nested and nested[0].isnumeric():
+                        expected_container_types.add(FormVariableDataTypes.array)
+
+                    if parent_var["type"] in expected_container_types:
+                        logger.info(
+                            "invalid variable reference (%s) in logic of form %s",
+                            var.key,
+                            form.admin_name,
+                        )
+                else:
+                    invalid_logic_rules.append(
+                        InvalidLogicRule(
+                            variable=var.key,
+                            form_name=form.admin_name,
+                            form_id=form.id,
+                        )
+                    )
```

Comment:
for var in set(form_logics_vars):
            # there is a variable with this exact key, it is a valid reference
            if var.key in all_keys:
                continue

            outer, *nested = var.key.split(".")

            def _report():
                invalid_logic_rules.append(
                    InvalidLogicRule(
                        variable=var.key,
                        form_name=form.admin_name,
                        form_id=form.id,
                    )
                )

            # Before checking the type of the parent/outer bit, check if it exists in
            # the first place. It could also be that it's not a parent at all (nested is
            # empty), so that's a guaranteed broken reference too.
            if outer not in all_keys:
                _report()
                continue

            # Nested cannot be empty now - if it were, either the key exists as a var,
            # which is the first thing we checked above, or it doesn't exist as a var,
            # which is the second thing we checked above and errored out.
            assert nested
            
            # We can only check the outer bit of the variable, which must be a suitable
            # container type to be valid. Container types are objects and arrays, but
            # we only consider arrays as a valid type if the first bit of the nested
            # key is numeric, as this must refer to an array index.
            # E.g. foo.bar -> only objects are possible, but foo.3 -> objects and arrays
            # are possible (you could have a dict with the string "3" as key").
            expected_container_types = {FormVariableDataTypes.object}
            if nested[0].isnumeric():
                expected_container_types.add(FormVariableDataTypes.array)

            # Now, check that the type of the container variable is as expected. If the
            # variable does *not* have any of the container types, then it's a 
            # guaranteed error since lookups inside primitives are not possible.
            parent_var = form_variables.get(outer) or static_variables.get(outer)
            if parent_var["type"] not in expected_container_types:
                _report()
                continue

            # all the other situations - we cannot (yet) conclude anything meaningful,
            # so instead, log information for our own insight into (typical) usage/
            # constructions.
            logger.info(
                "possible invalid variable reference (%s) in logic of form %s",
                var.key,
                form.admin_name,
            )